src/main/webapp/app/course/learning-paths/participate/learning-path-container.component.ts

```diff
@@ -0,0 +1,164 @@
+import { Component, Input, OnInit, ViewEncapsulation } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { faChevronLeft, faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import { Exercise } from 'app/entities/exercise.model';
+import { LectureUnit } from 'app/entities/lecture-unit/lectureUnit.model';
+import { Lecture } from 'app/entities/lecture.model';
+import { LearningPathService } from 'app/course/learning-paths/learning-path.service';
+import { NgxLearningPathNode, NodeType } from 'app/entities/competency/learning-path.model';
+import { LectureService } from 'app/lecture/lecture.service';
+import { onError } from 'app/shared/util/global.utils';
+import { HttpErrorResponse } from '@angular/common/http';
+import { AlertService } from 'app/core/util/alert.service';
+import { LearningPathLectureUnitViewComponent } from 'app/course/learning-paths/participate/lecture-unit/learning-path-lecture-unit-view.component';
+import { CourseExerciseDetailsComponent } from 'app/overview/exercise-details/course-exercise-details.component';
+import { ExerciseService } from 'app/exercises/shared/exercise/exercise.service';
+import { ExerciseEntry, LearningPathHistoryStorageService, LectureUnitEntry } from 'app/course/learning-paths/participate/learning-path-history-storage.service';
+
+@Component({
+    selector: 'jhi-learning-path-container',
+    styleUrls: ['./learning-path-container.component.scss'],
+    templateUrl: './learning-path-container.component.html',
+    encapsulation: ViewEncapsulation.None,
+})
+export class LearningPathContainerComponent implements OnInit {
+    @Input() courseId: number;
+    learningPathId: number;
+
+    learningObjectId: number;
+    lectureId?: number;
+    lecture: Lecture | undefined;
+    lectureUnit: LectureUnit | undefined;
+    exercise: Exercise | undefined;
```

Comment:
Can we keep these delcarations consistent?

```suggestion
    lecture?: Lecture;
    lectureUnit?: LectureUnit;
    exercise?: Exercise;
```